src/batou/utils.py

```diff
@@ -167,7 +167,8 @@ def __init__(self, connect_address, port=None):
             self.listen_v6 = NetLoc(v6_address, str(port))
 
         if not self.listen and not self.listen_v6:
-            raise socket.gaierror("No v4 or v6 address for %s" % connect)
+            raise socket.gaierror(
+                "No v4 or v6 address for '{}'.".format(connect))
```

Comment:
Hmm. Dang. I'm still getting confused. Can we get an example test with a message that shows the improved usability that you're working towards? I guess we should end up with an improper double quoting in that case, so the outer string with quotes around {} is fine but I suggest you use {!r} as the formatter:

```python
>>> print("Not a proper address {!r}".format("'asdf'"))
Not a proper address "'asdf'"
```

Then we und up with a double quoted string that will be (a bit more) obvious about the error the user made ...